plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java

```diff
@@ -477,8 +477,7 @@ public String toString() {
         @Override
         public void close() throws IOException {
             super.close();
-            if (in instanceof S3RetryingInputStream) {
-                final S3RetryingInputStream s3Stream = (S3RetryingInputStream) in;
+            if (in instanceof S3RetryingInputStream s3Stream) {
```

Comment:
In my related PRs I considered the `final [variable_name]` code as one that shouldn't change. The reason being that now the variable is not `final` anymore. My approach and suggestion is to keep the original code as is without `instanceof` pattern replacement.

Reply:
I see your point, I will follow your lead and keep the original code.

Reply:
I believe it should be possible to use `if (in instanceof final S3RetryingInputStream s3Stream) {`